### PR TITLE
Simplify on_demand mode.

### PR DESCRIPTION
**Pull request by maltoe**

### CODE REVIEW

File: lib/chromic_pdf/supervisor.ex [hunk 1]
```diff
@@ -26,69 +26,38 @@ defmodule ChromicPDF.Supervisor do
...
-      else
-        :supervisor
-      end
-
+  def child_spec(config) do
     %{
-      id: chromic,
-      start: {chromic, :start_link, [config]},
-      type: type
+      id: Keyword.fetch!(config, :name),
```
Comment (maltoe) [inline lib/chromic_pdf/supervisor.ex:40]: drive-by: also make the `id` dependent on the name to avoid conflicts when dynamic names are used

File: lib/chromic_pdf/supervisor.ex [hunk 2]
```diff
@@ -125,39 +84,38 @@ defmodule ChromicPDF.Supervisor do
...
-      {:found, fun.(pid)}
+  defp with_browser(supervisor, fun) do
+    if agent = find_supervisor_child(supervisor, Agent) do
+      with_on_demand_browser(agent, fun)
     else
-      :not_found
+      supervisor
+      |> find_supervisor_child!(Browser)
+      |> fun.()
     end
```
Comment (andreasknoepfle) [inline lib/chromic_pdf/supervisor.ex:103]: Wouldn't it make more sense to swap these around and rather find the `Browser` first, since that seems the more default use case?

**Review by andreasknoepfle — APPROVED** [review]
